src/lib.rs

```diff
@@ -100,12 +107,14 @@ struct Core {
 
 struct Slot<T> {
     value: UnsafeCell<MaybeUninit<T>>,
+    /// n-th bit is set if the n-th slot is being read from.
```

Comment:
It might be worth having a comment here that describes the use of the other bits in this field, as well, but since I didn't already write that down, I won't block this PR on making you do it for me. :)